pkg/util/crd/crd.go

```diff
@@ -0,0 +1,63 @@
+package crd
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	crdv1alpha1 "antrea.io/nephe/apis/crd/v1alpha1"
+	"github.com/cenkalti/backoff/v4"
+	"github.com/go-logr/logr"
+	apiextensionsclientset "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// DoesCesCrExistsForAccount returns true if there is a CloudEntitySelector CR for a given account.
+func DoesCesCrExistsForAccount(k8sClient client.Client, namespacedName *types.NamespacedName) bool {
+	cesList := &crdv1alpha1.CloudEntitySelectorList{}
+	listOptions := &client.ListOptions{
+		Namespace: namespacedName.Namespace,
+	}
+	if err := k8sClient.List(context.TODO(), cesList, listOptions); err != nil {
+		return false
+	}
+
+	for _, ces := range cesList.Items {
+		if ces.Spec.AccountName == namespacedName.Name {
+			return true
+		}
+	}
+	return false
+}
+
+// CheckCRDExistence checks if the custom resource definitions are installed on the API server.
+func CheckCRDExistence(log logr.Logger) error {
+	apiExtClient, err := apiextensionsclientset.NewForConfig(ctrl.GetConfigOrDie())
+	if err != nil {
+		return fmt.Errorf("failed to create API extensions client: %v", err)
+	}
+	b := backoff.NewExponentialBackOff()
+	// Configure the backoff parameters
+	b.InitialInterval = 1 * time.Second
+	b.MaxInterval = 30 * time.Second
+	b.MaxElapsedTime = 60 * time.Minute // Max retry
+
+	err = backoff.Retry(func() error {
+		crds := []string{
+			"cloudprovideraccounts.crd.cloud.antrea.io",
+			"cloudentityselectors.crd.cloud.antrea.io",
```

Comment:
nit: define these as const or slice var at top at package level?

Reply:
its actually in utility.. and relevant for this.. if it was dedicated package, then it make sense to put at the top. Hope it make sense